pkg/resources/storage_integration.go

```diff
@@ -0,0 +1,332 @@
+		stmt.SetString(`COMMENT`, v.(string))
+	}
+
+	if _, ok := data.GetOk("storage_blocked_locations"); ok {
+		stmt.SetStringList("STORAGE_BLOCKED_LOCATIONS", expandStringList(data.Get("storage_blocked_locations").([]interface{})))
+	}
+
+	// Now, set the storage provider
+	err := setStorageProviderSettings(data, stmt)
+	if err != nil {
+		return err
+	}
+
+	err = DBExec(db, stmt.Statement())
+	if err != nil {
+		return fmt.Errorf("error creating storage integration: %w", err)
+	}
+
+	data.SetId(name)
+
+	return ReadStorageIntegration(data, meta)
+}
+
+// ReadStorageIntegration implements schema.ReadFunc
+func ReadStorageIntegration(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	id := data.Id()
+
+	stmt := snowflake.StorageIntegration(data.Id()).Show()
+	row := db.QueryRow(stmt)
+
+	// Some properties can come from the SHOW INTEGRATION call
+	var name, integrationType, category, createdOn sql.NullString
+	var enabled sql.NullBool
+	if err := row.Scan(&name, &integrationType, &category, &enabled, &createdOn); err != nil {
+		return fmt.Errorf("Could not show storage integration: %w", err)
+	}
+
+	// Note: category must be STORAGE or something is broken
+	if c := category.String; c != "STORAGE" {
+		return fmt.Errorf("Expected %v to be a STORAGE integration, got %v", id, c)
+	}
+
+	if err := data.Set("name", name.String); err != nil {
+		return err
+	}
+
+	if err := data.Set("type", integrationType.String); err != nil {
+		return err
+	}
+
+	if err := data.Set("created_on", createdOn.String); err != nil {
+		return err
+	}
+
+	if err := data.Set("enabled", enabled.Bool); err != nil {
+		return err
+	}
+
+	// Some properties come from the DESCRIBE INTEGRATION call
+	// We need to grab them in a loop
+	var k, pType string
+	var v, d interface{}
+	stmt = snowflake.StorageIntegration(data.Id()).Describe()
+	rows, err := db.Query(stmt)
+	if err != nil {
+		return fmt.Errorf("Could not describe storage integration: %w", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		if err := rows.Scan(&k, &pType, &v, &d); err != nil {
+			return err
+		}
+		switch k {
+		case "STORAGE_PROVIDER":
+			if err = data.Set("storage_provider", v.(string)); err != nil {
+				return err
+			}
+		case "STORAGE_ALLOWED_LOCATIONS":
+			if err = data.Set("storage_allowed_locations", strings.Split(v.(string), ",")); err != nil {
+				return err
+			}
+		case "STORAGE_BLOCKED_LOCATIONS":
+			if val := v.(string); val != "" {
+				if err = data.Set("storage_blocked_locations", strings.Split(val, ",")); err != nil {
+					return err
+				}
+			}
+		case "STORAGE_AWS_IAM_USER_ARN":
+			if err = data.Set("storage_aws_iam_user_arn", v.(string)); err != nil {
+				return err
+			}
+		case "STORAGE_AWS_ROLE_ARN":
+			if err = data.Set("storage_aws_role_arn", v.(string)); err != nil {
+				return err
+			}
+		case "STORAGE_AWS_EXTERNAL_ID":
+			if err = data.Set("storage_aws_external_id", v.(string)); err != nil {
+				return err
+			}
+		}
```

Comment:
Add default statement here to error if `k` is not any of these values.

Reply:
An error would break the provider if Snowflake adds a new property here, so I have printed a warning instead.